frontends/p4/toP4/toP4.cpp

```diff
@@ -688,15 +689,18 @@ bool ToP4::preorder(const IR::Declaration_Variable *v) {
 
 bool ToP4::preorder(const IR::Type_Error *d) {
     dump(1);
+
     bool first = true;
     for (auto a : *d->getDeclarations()) {
-        if (ifSystemFile(a->getNode()).has_value())
+        if (isDeclaration && ifSystemFile(a->getNode()).has_value())
             // only print if not from a system file
             continue;
         if (!first) {
             builder.append(",\n");
         } else {
-            builder.append("error ");
+            builder.append("error");
+            if (!isDeclaration) return false;
```

Comment:
The logic seems to be a bit convoluted here. Can the printing of `error` be hoisted out of loop? Could it be possible the list of declarations is empty, so we'd essentially not print `error` here?

Reply:
Right. It will be better to handle this case outside of the for loop. Thanks for catching this.